src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V69/Eth69ProtocolHandler.cs

```diff
@@ -0,0 +1,110 @@
+using Nethermind.Blockchain;
+using Nethermind.Consensus;
+using Nethermind.Consensus.Scheduler;
+using Nethermind.Core;
+using Nethermind.Logging;
+using Nethermind.Network.Contract.P2P;
+using Nethermind.Network.P2P.Subprotocols.Eth.V62;
+using Nethermind.Network.P2P.Subprotocols.Eth.V63;
+using Nethermind.Network.P2P.Subprotocols.Eth.V66.Messages;
+using Nethermind.Network.P2P.Subprotocols.Eth.V68;
+using Nethermind.Network.P2P.Subprotocols.Eth.V69.Messages;
+using Nethermind.Network.Rlpx;
+using Nethermind.Stats;
+using Nethermind.Synchronization;
+using Nethermind.TxPool;
+
+namespace Nethermind.Network.P2P.Subprotocols.Eth.V69;
+
+/// <summary>
+/// https://github.com/MariusVanDerWijden/EIPs/blob/eth69_v2/EIPS/eip-7642.md
+/// </summary>
+public class Eth69ProtocolHandler : Eth68ProtocolHandler
+{
+    public Eth69ProtocolHandler(ISession session,
+        IMessageSerializationService serializer,
+        INodeStatsManager nodeStatsManager,
+        ISyncServer syncServer,
+        IBackgroundTaskScheduler backgroundTaskScheduler,
+        ITxPool txPool,
+        IPooledTxsRequestor pooledTxsRequestor,
+        IGossipPolicy gossipPolicy,
+        ForkInfo forkInfo,
+        ILogManager logManager,
+        ITxGossipPolicy? transactionsGossipPolicy = null)
+        : base(session, serializer, nodeStatsManager, syncServer, backgroundTaskScheduler, txPool, pooledTxsRequestor, gossipPolicy, forkInfo, logManager, transactionsGossipPolicy)
+    {
+    }
+
+    public override string Name => "eth69";
+
+    public override byte ProtocolVersion => EthVersions.Eth69;
+
+    public override void HandleMessage(ZeroPacket message)
+    {
+        int size = message.Content.ReadableBytes;
+        switch (message.PacketType)
+        {
+            case Eth62MessageCode.NewBlockHashes:
+                break;
+            case Eth62MessageCode.NewBlock:
+                break;
+            case Eth62MessageCode.Status:
+                StatusMessage69 statusMsg = Deserialize<StatusMessage69>(message.Content);
+                base.ReportIn(statusMsg, size);
+                this.Handle(statusMsg);
+                break;
+            case Eth63MessageCode.Receipts:
+                ReceiptsMessage69 receiptsMessage = Deserialize<ReceiptsMessage69>(message.Content);
+                base.ReportIn(receiptsMessage, size);
+                base.Handle(receiptsMessage, size);
+                break;
+            case Eth63MessageCode.GetReceipts:
+                GetReceiptsMessage getReceiptsMessage = Deserialize<GetReceiptsMessage>(message.Content);
+                ReportIn(getReceiptsMessage, size);
+                BackgroundTaskScheduler.ScheduleSyncServe(getReceiptsMessage, this.Handle);
+                break;
+            default:
+                base.HandleMessage(message);
+                break;
+        }
+    }
+
+    private void Handle(StatusMessage69 status)
+    {
+        status.TotalDifficulty = 0; // TODO handle properly for eth/69
```

Comment:
This value is later assigned to current `ISyncPeer` `TotalDifficulty`, which has a lot of usages. While it looks like none of them should actually apply in the post-merge chains with code branching elsewhere, to fully confirm, one would need to do additional manual/integration testing, for example do a full sync from one node to another using the new protocol version.

Another potential problem, is someone using eth/69 with a non-PoS chain, for which `TotalDifficulty` is actually being needed for syncing.

Reply:
@LukaszRozmej where do you prefer to use `TerminalTotalDifficulty`? `Status` message of `ProtocolHandlerBase` don't have such property.

This assignment is here just because handler doesn't support nullable `TotalDifficulty` and changing it will affect a lot of files.